includes/Register_Widget.php

```diff
@@ -58,17 +60,51 @@ public function __construct( Stories $stories ) {
 	 * @return void
 	 */
 	public function register() {
+		add_action( 'widgets_init', [ $this, 'register_widget' ] );
+		add_filter( 'widget_types_to_hide_from_legacy_widget_block', [ $this, 'hide_widget' ] );
+		add_filter( 'body_class', [ $this, 'body_class' ] );
+	}
+
+	/**
+	 * Register widget.
+	 *
+	 * @since 1.9.0
+	 *
+	 * @return void
+	 */
+	public function register_widget() {
 		register_widget( $this->stories );
 	}
 
 	/**
-	 * Get the action to use for registering the service.
+	 * Hide widget stories from legacy widget list.
 	 *
-	 * @since 1.6.0
+	 * @since 1.9.0
+	 *
+	 * @param array $widget_types An array of excluded widget-type IDs.
 	 *
-	 * @return string Registration action to use.
+	 * @return array
 	 */
-	public static function get_registration_action() {
-		return 'widgets_init';
+	public function hide_widget( array $widget_types ) {
+		$widget_types[] = $this->stories->id_base;
+
+		return $widget_types;
+	}
+
+	/**
+	 * Filters the list of CSS body class names for embedded iframes to add a class.
+	 *
+	 * @since 1.9.0
+	 *
+	 * @param string[] $classes An array of body class names.
+	 *
+	 * @return array
+	 */
+	public function body_class( array $classes ) {
+		if ( is_admin() && defined( 'IFRAME_REQUEST' ) && IFRAME_REQUEST ) {
+			$classes[] = 'iframe-embed';
```

Comment:
Can we rename this to something like `ws-legacy-widget-preview`?

Reply:
This would allow to all legacy widget iframe preview. Does this naming make sense?

Reply:
`iframe-embed` is just too generic. It should be clear that it's added by us and should only be used by us. Even if it happens to apply to all previews.